### PR TITLE
Add missing Aspect Ratio query parameter

### PR DESCRIPTION
**Pull request by PeterDekkers**
Adds [Aspect Ratio](https://docs.imgix.com/apis/url/size/ar) to the `ImgixUrlQueryParams` type.

Adds validation of this parameter to ensure that a proper aspect ratio string is passed.

### CODE REVIEW

**Review by OliverJAsh — COMMENTED** [review]
Thanks for your PR!

I would like to try and improve type safety so that more errors can be caught at compile time. Currently it would be possible for a user to specify `ar: 'foo'` and this won't error until runtime validation happens.

If we instead expressed this parameter as type of `{ w: number; h: number }`, this sort of error would be caught at compile time.

We would then convert this object to the string format required by imgix (inside `serializeImgixUrlQueryParamValues`).

Another idea: [the docs](https://docs.imgix.com/apis/url/size/ar) specify:

> This parameter will work only when `fit=crop` is set.

We could enforce this at the type level, to help catch more mistakes at compile time.

**Comment by PeterDekkers** [comment]
Thanks @OliverJAsh, great suggestions. Will make a few changes later on.

**Comment by PeterDekkers** [comment]
Much better with the typed number parameters.

I wasn't sure how to enforce a check for presence of `fit=crop` at the type level, sorry.

File: src/index.ts [hunk 1]
```diff
@@ -80,6 +87,7 @@ type ParsedUrlQueryInput = { [key: string]: unknown };
 const serializeImgixUrlQueryParamValues = (query: ImgixUrlQueryParams): ParsedUrlQueryInput =>
     pipe(
         (): Record<keyof ImgixUrlQueryParams, string | number | undefined> => ({
+            ar: typeof query.ar !== 'undefined' ? `${query.ar.w}:${query.ar.h}` : undefined,
```
Comment (OliverJAsh) [inline src/index.ts]: I think you can use `mapValueIfDefined` here, e.g.

```ts
ar: mapValueIfDefined(ar => `${ar.w}:${ar.h}`)
```

**Review by OliverJAsh — CHANGES_REQUESTED** [review]
> I wasn't sure how to enforce a check for presence of `fit=crop` at the type level, sorry.

No problem, it would be fiddly. Let's leave it for now. I've opened an issue (https://github.com/unsplash/ts-imgix/issues/6) to track the suggestion.

File: src/index.ts (hunk 1 above)
Comment (PeterDekkers) [reply src/index.ts]: Ah yep, adjustment made. Thank you.

**Comment by PeterDekkers** [comment]
Makes sense to pull the `fit=crop` check out to a separate issue. Will keep pondering it and, who knows, might submit a PR for it in the near future. Thanks!

File: src/index.ts (hunk 1 above)
Comment (PeterDekkers) [reply src/index.ts]: I have to mention while I'm here that I found the original plain vanilla JS approach more legible, easier to comprehend. :v:

**Comment by OliverJAsh** [comment]
Published as [v0.0.7](https://github.com/unsplash/ts-imgix/releases/tag/v0.0.7)

**Pull request merged**